src/sentry/management/commands/backfill_eventstream.py

```diff
@@ -0,0 +1,90 @@
+
+:copyright: (c) 2018 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+from __future__ import absolute_import, print_function
+
+import sys
+
+from django.core.management.base import BaseCommand, CommandError, make_option
+from django.utils.dateparse import parse_datetime
+
+from sentry.models import Event
+
+
+class Command(BaseCommand):
+    help = 'Backfill events from the database into the event stream.'
+
+    option_list = BaseCommand.option_list + (
+        make_option('--from-ts', dest='from_ts', type='string',
+                    help='Starting event timestamp (ISO 8601). Example: 2018-11-26T23:59:59'),
+        make_option('--to-ts', dest='to_ts', type='string',
+                    help='Last event timestamp (ISO 8601).'),
+        make_option('--from-id', dest='from_id', type=int,
+                    help='Starting event ID (primary key).'),
+        make_option('--to-id', dest='to_id', type=int,
+                    help='Last event ID (primary key).'),
+        make_option('--no-input', action='store_true', dest='no_input',
+                    help='Do not ask questions.')
+    )
+
+    def get_events_by_timestamp(self, from_ts, to_ts):
+        from_date = parse_datetime(from_ts)
+        if not from_date:
+            raise CommandError('Cannot parse --from-ts')
+        to_date = parse_datetime(to_ts)
+        if not to_date:
+            raise CommandError('Cannot parse --to-ts')
+        return Event.objects.filter(datetime__gte=from_date, datetime__lte=to_date)
+
+    def get_events_by_id(self, from_id, to_id):
+        if from_id > to_id:
+            raise CommandError('Invalid ID range.')
+        return Event.objects.filter(id__gte=from_id, id__lte=to_id)
+
+    def handle(self, **options):
+        from sentry import eventstream
+
+        from_ts = options['from_ts']
+        to_ts = options['to_ts']
+        from_id = options['from_id']
+        to_id = options['to_id']
+
+        if (from_ts or to_ts) and (from_id or to_id):
+            raise CommandError('You can either limit by primary key, or by timestamp.')
+        elif from_ts and to_ts:
+            events = self.get_events_by_timestamp(from_ts, to_ts)
+        elif from_id and to_id:
+            events = self.get_events_by_id(from_id, to_id)
+        else:
+            raise CommandError('Invalid arguments: either use --from/--to-id, or --from/--to-ts.')
+
+        count = events.count()
+        self.stdout.write('Events to process: {}'.format(count))
+
+        if count == 0:
+            self.stdout.write('Nothing to do.')
+            sys.exit(0)
+
+        if not options['no_input']:
+            proceed = raw_input('Do you want to continue? [y/N] ')
+            if proceed.lower() not in ['yes', 'y']:
+                raise CommandError('Aborted.')
+
+        for event in events:
+            primary_hash = event.get_primary_hash()
+            eventstream.insert(
```

Comment:
Inside this call we'll be hammering project lookups:

https://github.com/getsentry/sentry/blob/772c0fb32b4ce944568fc6587c1a68b92956cf9f/src/sentry/eventstream/kafka.py#L123

We may want to make sure that's OK with production or else use `select_related` or one of the smarter "attach FK" methods (I know we have something in the codebase but I can't remember the name).

Reply:
I would select related. We have a similar situation with the `group` a bit below.

Reply:
`group` is only used in the base class, which is skipped because of `skip_consume=True` so we could even set `group=False` but that's definitely a hack...